controllers/provider-aws/pkg/controlplane/internal/helper.go

```diff
@@ -0,0 +1,235 @@
+}
+
+// NewHelper creates and returns a new Helper instance.
+func NewHelper() *Helper {
+	return &Helper{
+		Secrets:                     controlPlaneSecrets,
+		ConfigChart:                 configChart,
+		ControlPlaneChart:           controlPlaneChart,
+		ImageVectorFunc:             awsimagevector.GetImageVector,
+		ConfigChartValuesFunc:       getConfigChartValues,
+		ControlPlaneChartValuesFunc: getControlPlaneChartValues,
+	}
+}
+
+var (
+	controlPlaneSecrets = &secrets.Secrets{
+		CertificateSecretConfigs: map[string]*secrets.CertificateSecretConfig{
+			gardencorev1alpha1.SecretNameCACluster: {
+				Name:       gardencorev1alpha1.SecretNameCACluster,
+				CommonName: "kubernetes",
+				CertType:   secrets.CACert,
+			},
+		},
+		SecretConfigsFunc: func(cas map[string]*secrets.Certificate, clusterName string) []secrets.ConfigInterface {
+			return []secrets.ConfigInterface{
+				&secrets.ControlPlaneSecretConfig{
+					CertificateSecretConfig: &secrets.CertificateSecretConfig{
+						Name:         cloudControllerManagerDeploymentName,
+						CommonName:   "system:cloud-controller-manager",
+						Organization: []string{user.SystemPrivilegedGroup},
+						CertType:     secrets.ClientCert,
+						SigningCA:    cas[gardencorev1alpha1.SecretNameCACluster],
+					},
+					KubeConfigRequest: &secrets.KubeConfigRequest{
+						ClusterName:  clusterName,
+						APIServerURL: common.KubeAPIServerDeploymentName,
+					},
+				},
+				&secrets.ControlPlaneSecretConfig{
+					CertificateSecretConfig: &secrets.CertificateSecretConfig{
+						Name:       cloudControllerManagerServerName,
+						CommonName: cloudControllerManagerDeploymentName,
+						DNSNames:   controlplane.DNSNamesForService(cloudControllerManagerDeploymentName, clusterName),
+						CertType:   secrets.ServerCert,
+						SigningCA:  cas[gardencorev1alpha1.SecretNameCACluster],
+					},
+				},
+			}
+		},
+	}
+
+	configChart = &chart.Chart{
+		Name: "cloud-provider-config",
+		Path: filepath.Join(aws.InternalChartsPath, "cloud-provider-config"),
+		Objects: []*chart.Object{
+			{Type: &corev1.ConfigMap{}, Name: common.CloudProviderConfigName},
+		},
+	}
+
+	controlPlaneChart = &chart.Chart{
+		Name:   "cloud-controller-manager",
+		Path:   filepath.Join(aws.InternalChartsPath, "cloud-controller-manager"),
+		Images: []string{common.HyperkubeImageName},
+		Objects: []*chart.Object{
+			{Type: &corev1.Service{}, Name: "cloud-controller-manager"},
+			{Type: &appsv1.Deployment{}, Name: "cloud-controller-manager"},
+		},
+	}
+)
+
+// getConfigChartValues collects and returns the configuration chart values.
+func getConfigChartValues(decoder runtime.Decoder, cp *extensionsv1alpha1.ControlPlane, cluster *extensionscontroller.Cluster) (map[string]interface{}, error) {
+	// Decode providerConfig
+	cpConfig := &apisaws.ControlPlaneConfig{}
```

Comment:
Same as below, write a decoding function with a constant scheme and make this method operate on the decoded variant.